lib/developer_portal/lib/cms/toolbar.rb

```diff
@@ -39,6 +39,28 @@ def draft?
 
   private
 
+  def validate_and_extract_cms_token
+    signature = params.delete(:signature)
+    return signature if signature.blank?
+
+    expires_at = Time.at(params.delete(:expires_at).to_i).utc
+    raise ActiveSupport::MessageVerifier::InvalidSignature unless expires_at > Time.now.utc
+
+    verifier = Rails.application.message_verifier(:cms_token)
```

Comment:
Would it maybe make sense to extract these duplicated lines to a method, and call this method both from here to validate the token, and from the controller that generates it?

Reply:
Yeah, only if extracting these out doesn't make a bigger mess. DRY not to the extreme :)